dev/tests/static/framework/tests/unit/testsuite/Magento/TestFramework/Utility/ChildrenClassesSearchTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\TestFramework\Utility;
+
+use Magento\Framework\App\Action\AbstractAction;
+
+class ChildrenClassesSearchTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var ChildrenClassesSearch
+     */
+    private $childrenClassesSearch;
+
+    protected function setUp(): void
+    {
+        $this->childrenClassesSearch = new ChildrenClassesSearch();
+    }
+
+    public function testChildrenSearch(): void
```

Comment:
This method suppose to run not only against action interface and abstract action.

Would be good to add additional check against some clean "A" class to make sure it will work also for other cases:
class A {} -> ❌
class B extends A {} -> ✔
class C implements Z {} -> ❌ (not extends from A)
class D {} -> ❌ (not extends from A)
class E extends B {} -> ✔(as E extends B, B extends A)
class E extends B implements Z {} -> ✔(as E extends B, B extends A)

Reply:
done